src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -2798,6 +2785,9 @@ private ITaskItem SetItemMetadata(List<ITaskItem> relatedItems, List<ITaskItem>
                 }
             }
 
+            // Set the FusionName metadata properly.
+            referenceItem.SetMetadata(ItemMetadataNames.fusionName, fusionName);
```

Comment:
If any of reference.GetSourceItems() has a fusionName, or reference.PrimarySourceItem has one, fusionName may have been set to an invalid value at this point. We remove it before forwarding referenceItem's metadata to other items, but we still need to set it properly at the end, i.e., give it the right fusionName.

Reply:
Would you mind clarifying what you meant about fusionName being guaranteed to be empty? Previously, fusionName was set to a value near the top of this method, that value was passed on to a variety of "relatedItems," "satelliteItems," etc., and then it was unset from all of those other items. Here, instead, I'm unsetting it prior to using fusionName for any of those, and only giving it its value at the bottom, so CopyMetadataTo automatically skips fusionName.

Reply:
The meta comment is: "properly" is basically useless in a code comment. Comments describe what you want the code to do/why it's done, and of course you _want_ to do things "properly". Instead, describe details. Here, you might have a comment that `fusionName` should be set last, so that it doesn't get copied to the derived items but is still available on the primary item.